improver_tests/nbhood/square_kernel/test_SquareNeighbourhood.py

```diff
@@ -78,47 +77,49 @@ class Test_run(IrisTest):
 
     RADIUS = 2500
 
+    def setUp(self):
+        """Create test cubes"""
+
+        # Create 2D test cube
+
+        data = np.ones((5, 5), dtype=np.float32)
+        data[2, 2] = 0
+        self.cube = set_up_variable_cube(data, spatial_grid="equalarea",)
+
+        # Create 3D test cube with 2 realizations
+
+        data = np.ones((2, 5, 5), dtype=np.float32)
+        self.multi_cube = set_up_variable_cube(data, spatial_grid="equalarea",)
+
     def test_basic_re_mask_true(self):
         """Test that a cube with correct data is produced by the run method
         when re-masking is applied."""
-        data = np.array(
+        expected_array = np.array(
             [
-                [
-                    [
-                        [1.0, 1.0, 1.0, 1.0, 1.0],
-                        [1.0, 0.88888889, 0.88888889, 0.88888889, 1.0],
-                        [1.0, 0.88888889, 0.88888889, 0.88888889, 1.0],
-                        [1.0, 0.88888889, 0.88888889, 0.88888889, 1.0],
-                        [1.0, 1.0, 1.0, 1.0, 1.0],
-                    ]
-                ]
+                [1.0, 1.0, 1.0, 1.0, 1.0],
+                [1.0, 0.88888889, 0.88888889, 0.88888889, 1.0],
+                [1.0, 0.88888889, 0.88888889, 0.88888889, 1.0],
+                [1.0, 0.88888889, 0.88888889, 0.88888889, 1.0],
+                [1.0, 1.0, 1.0, 1.0, 1.0],
             ]
         )
-        cube = set_up_cube(
-            zero_point_indices=((0, 0, 2, 2),), num_time_points=1, num_grid_points=5
-        )
+        cube = self.cube
         result = SquareNeighbourhood().run(cube, self.RADIUS)
         self.assertIsInstance(cube, Cube)
```

Comment:
Oh dear... although this isn't a problem you introduced, there are several places this crops up.  Could you please correct these `assert` statements to test the result and not the input cube!

Reply:
Done, and removed all except the first.

Reply:
👍